com.unity.perception/Runtime/GroundTruth/InstanceIdToColorMapping.cs

```diff
@@ -142,17 +148,52 @@ public static bool TryGetColorFromInstanceId(uint id, out Color32 color)
         }
 
         /// <summary>
-        /// Retrieve the ID associated with the passed in color. If the passed in color is black, or has an alpha channel
-        /// of 0, or has an alpha channel of value of 255, but the RGB values are not associated with a known color, this
-        /// service will return false, and the out id will be set to 0.
+        /// Retrieve the color that is mapped to the passed in ID. If the ID is 0 or 255 the returned color will be black.
+        /// </summary>
+        /// <param name="id">The ID of interest.</param>
+        /// <returns>The color associated with the passed in ID, or black if no associated color exists.</returns>
+        /// <exception cref="IndexOutOfRangeException">Thrown if the passed in ID is greater than the largest supported ID <see cref="MaxId"/></exception>
+        public static Color32 GetColorFromInstanceId(uint id)
+        {
+            if (id > MaxId) throw new IndexOutOfRangeException($"Passed in index: {id} is greater than max ID: {MaxId}");
+
+            var color = k_InvalidColor;
+            var packed = GetColorForId(id);
+
+            if (packed != k_InvalidPackedColor)
+                color = GetColorFromPackedColor(packed);
+
+            return color;
+        }
+
+        /// <summary>
+        /// Retrieve the ID associated with the passed in color. If the passed in color is black this service will return
+        /// false, and the out id will be set to 0.
         /// </summary>
         /// <param name="color">The color to map to an ID.</param>
         /// <param name="id">This value will be updated with the ID for the passed in color.</param>
         /// <returns>This service will return true if an ID is properly mapped to a color, otherwise it will return false.</returns>
+        /// <exception cref="IndexOutOfRangeException">Thrown if the passed in color is mapped to an ID that is greater than the largest supported ID</exception>
```

Comment:
It's possible for a TryGet* method to throw exceptions, but I'm not sure in this case. Give these guidelines a look to see if we should be throwing in this method https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/exceptions-and-performance#try-parse-pattern

Reply:
ok, TryGets no longer throw exceptions